### PR TITLE
Add test for jemalloc pool manager

### PR DESCRIPTION
**Pull request by ldorau**
Requires:
- [x] #64
- [x] #74

### CODE REVIEW

**Review by pbalcer — APPROVED** [review]

File: test/memoryPoolAPI.cpp
Comment (vinser52) [inline test/memoryPoolAPI.cpp:1]: As I understand we will execute the general pool manager tests we have so far for all other pool managers.
Should we test that jemalloc-specific API is implemented in the right way? I mean the case with split/merge arena extents.

**Review by KFilipek — APPROVED** [review]

File: test/memoryPoolAPI.cpp
Comment (ldorau) [reply test/memoryPoolAPI.cpp:1]: > As I understand we will execute the general pool manager tests we have so far for all other pool managers. 
>
Yes.

> Should we test that jemalloc-specific API is implemented in the right way? I mean the case with split/merge arena extents.
>
I think we should not test directly the internal jemalloc hooks. They are called many times (also the `arena_extent_split/merge` hooks) when we call the public pool's API, so they are already tested indirectly.

**Review by igchor — APPROVED** [review]

File: test/memoryPoolAPI.cpp
Comment (ldorau) [reply test/memoryPoolAPI.cpp:1]: If we implement splitting/merging extents, we can inverse the return value. Now it is not implemented, so we have to return a failure according to the reality ;-)

Comment (ldorau) [reply test/memoryPoolAPI.cpp:1]: > So, in other words: if we inverse the return value of the `arena_extent_split/merge` functions the existing tests will catch that?

The existing tests already call those extents, so they can fail if you inverse the return value without implementing them.

**Review by vinser52 — APPROVED** [review]

**Pull request merged**